### ISSUE TITLE
Merge properly into current master before testing

### ISSUE THREAD

**Issue by mxinden** [coreos/tectonic-installer#1625]
The [`checkout scm`](https://github.com/coreos/tectonic-installer/blob/master/Jenkinsfile#L464) step, that we use to check out the git revision on the worker nodes, will only checkout the revision itself. It will not merge it into master before starting the test execution.

In my eyes this is quite error prone. We should follow a *continuous integration* (into master) approach, like pointed out by @Quentin-M. This will ensure, that the changes of a PR are fully compatible with the master branch.

For further details see https://github.com/coreos/tectonic-installer/pull/1611, where this came up.

**Comment by cpanato** [comment]
I think we can add the strategy during the checkout to merge to master.

I will check

**Comment by mxinden** [comment]
Closing in favour of https://coreosdev.atlassian.net/browse/INST-78
